### ISSUE TITLE
A question about augmentation yolov5/utils/augmentations.py and yolov5/data/hyp.*.yaml files

### ISSUE THREAD

**Issue by frabob2017** [ultralytics/yolov5#9761]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I check the  yolov5/utils/augmentations.py. It looks that by default, there is a rotation of 10 degree for data augmentation. However, in yolov5/data/hyp.*.yaml files, I found that the rotation degree is 0 degree. In this situation, which one determine rotation will be used for augmentation?  If I want to change rotation degree to 20 degree, do I need to change  yolov5/utils/augmentations.py to set degrees=20 or how can I change it in command line? 

def random_perspective(im,
                       targets=(),
                       segments=(),
                       degrees=10,
                       translate=.1,
                       scale=.1,
                       shear=10,
                       perspective=0.0,
                       border=(0, 0)):

**Comment by glenn-jocher** [comment]
@frabob2017 hyp file sets rotation, yes you can just change to 20 deg and it will apply to training.

**Comment by frabob2017** [comment]
> @frabob2017 hyp file sets rotation, yes you can just change to 20 deg and it will apply to training.

There are multiple  yolov5/data/hyp.*.yaml files, which one should I use?   I prefer to use  yolov5/utils/augmentations.py?  How to use this? I set degrees=20 directly in this file?   

![image](https://user-images.githubusercontent.com/57141860/195198724-fa53717e-022c-4a94-9172-ee388cdb7b78.png)

**Comment by glenn-jocher** [comment]
You can see the default hyp file (low) used in train.py argparser. Higher augmentation reduces overfitting.

**Comment by frabob2017** [comment]
> You can see the default hyp file (low) used in train.py argparser. Higher augmentation reduces overfitting.

Is there any tutorials how to use yolov5/utils/augmentations.py, it seems that you recommend to use yolov5/data/hyp.*.yaml.

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about **image augmentation**. YOLOv5 🚀 applies online imagespace and colorspace augmentations in the trainloader (but not the val_loader) to present a new and unique augmented [Mosaic](https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/utils/datasets.py#L678-L732) (original image + 3 random images) each time an image is loaded for training. **Images are never presented twice in the same way**.

![YOLOv5 augmentation](https://user-images.githubusercontent.com/26833433/120995721-f3cfed00-c785-11eb-8ee2-b6ef2fa205e8.jpg)

### Augmentation Hyperparameters

The hyperparameters used to define these augmentations are in your hyperparameter file (default `data/hyp.scratch.yaml`) defined when training:
```
python train.py --hyp hyp.scratch-low.yaml
```
https://github.com/ultralytics/yolov5/blob/b94b59e199047aa8bf2cdd4401ae9f5f42b929e6/data/hyps/hyp.scratch-low.yaml#L6-L34

### Augmentation Previews

You can view the effect of your augmentation policy in your train_batch*.jpg images once training starts. These images will be in your train logging directory, typically `yolov5/runs/train/exp`:

`train_batch0.jpg` shows train batch 0 mosaics and labels:  
<img src="https://user-images.githubusercontent.com/26833433/131255960-b536647f-7c61-4f60-bbc5-cb2544d71b2a.jpg" width="800">


### YOLOv5 Albumentations Integration

[YOLOv5](https://github.com/ultralytics/yolov5) 🚀 is now fully integrated with [Albumentations](https://github.com/albumentations-team/albumentations), a popular open-source image augmentation package. Now you can train the world's best Vision AI models even better with custom Albumentations 😃! 

PR https://github.com/ultralytics/yolov5/pull/3882 implements this integration, which will automatically apply Albumentations transforms during YOLOv5 training if `albumentations>=1.0.3` is installed in your environment. See https://github.com/ultralytics/yolov5/pull/3882 for full details.

Example `train_batch0.jpg` on COCO128 dataset with Blur, MedianBlur and ToGray. See the YOLOv5 [Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) to reproduce: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

<img src="https://user-images.githubusercontent.com/26833433/124400879-ff331b80-dd25-11eb-9b67-fe85ac4ca104.jpg" width="800">

Good luck 🍀 and let us know if you have any other questions!
